iib/workers/tasks/opm_operations.py

```diff
@@ -406,15 +406,91 @@ def opm_generate_dockerfile(
             log.info('Rewriting Dockerfile %s with newly generated by opm.', dockerfile_path)
         os.rename(dockerfile_path_opm_default, dockerfile_path)
 
+    local_cache_path = os.path.join(base_dir, 'cache')
+    generate_cache_locally(base_dir, fbc_dir, local_cache_path)
+    insert_cache_into_dockerfile(dockerfile_path, local_cache_path)
+
     db_path = get_worker_config()['hidden_index_db_path']
     rel_path_index_db = os.path.relpath(index_db, base_dir)
-    with open(dockerfile_path, 'a') as f:
+    with open(dockerfile_path, 'a+') as f:
         f.write(f'\nADD {rel_path_index_db} {db_path}\n')
+        verify_cache_insertion_into_dockerfile(f.readlines(), local_cache_path)
 
     log.info("Dockerfile was successfully generated.")
     return dockerfile_path
 
 
+def verify_cache_insertion_into_dockerfile(file_list: list, local_cache_path: str) -> None:
+    """
+    Verify local cache was copied into the dockerfile.
```

Comment:
Nitpick: We are not verifying if it was copied, we verify if the COPY command is present in Dockerfile.

Reply:
nice catch! changed it :)